File: api/src/endpoint/task/update_task.rs
use api_derive::ApiRequest;
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::TaskResource};

pub struct UpdateTask;

impl Endpoint for UpdateTask {
    const PATH: &'static str = "/ci/task/{id}";
    const METHOD: http::Method = http::Method::PATCH;

    type Request = UpdateTaskRequest;
    type Response = UpdateTaskResponse;
}

#[derive(ApiRequest, Debug, Clone, Serialize, Deserialize)]
pub struct UpdateTaskRequest {
    pub status: String,
}

pub type UpdateTaskResponse = TaskResource;

File: gitdot-api-ts/src/endpoint/task/update-task.ts
import { z } from "zod";

import { TaskResource } from "../../resource";

export const UpdateTaskRequest = z.object({
  status: z.string(),
});
export type UpdateTaskRequest = z.infer<typeof UpdateTaskRequest>;

export const UpdateTask = {
  path: "/ci/task/{id}",
  method: "PATCH",
  request: UpdateTaskRequest,
  response: TaskResource,
} as const;
export type UpdateTask = typeof UpdateTask;
